scapy/packet.py

```diff
@@ -555,6 +555,28 @@ def pdfdump(self, filename=None, **kargs):
             canvas.writePDFfile(filename)
         print()
 
+    def svgdump(self, filename=None, **kargs):
+        """
+        svgdump(filename=None, layer_shift=0, rebuild=1)
+
+        Creates an SVG file describing a packet. If filename is not provided a
+        temporary file is created and gs is called.
+
+        :param filename: the file's filename
+        """
+        canvas = self.canvas_dump(**kargs)
+        if filename is None:
+            fname = get_temp_file(autoext=".svg")
+            canvas.writeSVGfile(fname)
+            if WINDOWS and conf.prog.svgreader is None:
+                os.startfile(fname)
+            else:
+                with ContextManagerSubprocess("svgdump()", conf.prog.svgreader):  # noqa: E501
```

Comment:
please do not add `noqa` tags, fix the code instead. Maybe something like this would work?
```Python
                with ContextManagerSubprocess("svgdump()",
                                                                                conf.prog.svgreader):
```